sessionctx/variable/session.go

```diff
@@ -1281,6 +1281,68 @@ type SessionVars struct {
 	// OptPrefixIndexSingleScan indicates whether to do some optimizations to avoid double scan for prefix index.
 	// When set to true, `col is (not) null`(`col` is index prefix column) is regarded as index filter rather than table filter.
 	OptPrefixIndexSingleScan bool
+
+	MaxReuseChunk  int
+	MaxReuseColumn int
+	// ChunkPool Several chunks and columns are cached
+	ChunkPool struct {
+		Lock  sync.Mutex
+		Alloc chunk.Allocator
+	}
+	EnableReuseCheck bool
+
+	// UseChunkAlloc indicates whether statement use chunk alloc
+	UseChunkAlloc    bool
+	PreUseChunkAlloc bool
+}
+
+// GetNewChunk Attempt to request memory from the chunk pool
+// thread safety
+func (s *SessionVars) GetNewChunk(fields []*types.FieldType, capacity int) *chunk.Chunk {
+	//Chunk memory pool is not set
+	if s.ChunkPool.Alloc == nil {
+		return chunk.NewChunkWithCapacity(fields, capacity)
+	}
+	s.ChunkPool.Lock.Lock()
+	defer s.ChunkPool.Lock.Unlock()
+	if (s.MaxReuseChunk > 0 || s.MaxReuseColumn > 0) && (!s.UseChunkAlloc) {
+		s.UseChunkAlloc = true
+	}
+	chk := s.ChunkPool.Alloc.Alloc(fields, capacity, capacity)
+	return chk
+}
+
+// GetNewChunkWithCapacity Attempt to request memory from the chunk pool
+// thread safety
+func (s *SessionVars) GetNewChunkWithCapacity(fields []*types.FieldType, capacity int, maxCachesize int) *chunk.Chunk {
+	if s.ChunkPool.Alloc == nil {
+		return chunk.New(fields, capacity, maxCachesize)
+	}
+	s.ChunkPool.Lock.Lock()
+	defer s.ChunkPool.Lock.Unlock()
+	if s.MaxReuseChunk > 0 || s.MaxReuseColumn > 0 && (!s.UseChunkAlloc) {
+		s.UseChunkAlloc = true
```

Comment:
Is the variable necessary? Maybe you can defile a function?

```
func (s *SessionVars) UseChunkAlloc() bool {
    return s.MaxReuseChunk > 0 ||  s.MaxReuseColumn > 0
}
```

The different is that when you have several variable for something, and the value of some variables are derived from the others, it's easy to get into consistency between those variables.

Reply:
ok

Reply:
done